### PR TITLE
MonkeyZoo 

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature / Fixes
> Adds terraform files that can be used to deploy monkey's testing network on GCP.
> Adds documentation about the network and its deplyment

- [x] Tested on my own GCP account

### CODE REVIEW

**Comment by itaymm** [comment]
Move MonkeyZoo to /envs/monkey_zoo

File: MonkeyZoo/main.tf [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+      image = "${data.google_compute_image.hadoop-2.self_link}"
+    }
+    auto_delete = true
+  }
+  network_interface {
+    subnetwork="monkeyzoo-main"
+    network_ip="10.2.2.2"
+  }
+  // Add required ssh keys for hadoop service and restart it
+  metadata_startup_script = "[ ! -f /home/vakaris_zilius/.ssh/authorized_keys ] && sudo cat /home/vakaris_zilius/.ssh/id_rsa.pub >> /home/vakaris_zilius/.ssh/authorized_keys && sudo reboot"
```
Comment (danielguardicore) [inline MonkeyZoo/main.tf]: Hardcoded paths, won't work

File: envs/monkey_zoo/configs/fullTest.conf [hunk 2]
```diff
@@ -62,7 +62,7 @@
       "exploiter_classes": [
         "SmbExploiter",
         "WmiExploiter",
-        "SSHExploiter",
+        "RdpExploiter",
```
Comment (danielguardicore) [inline envs/monkey_zoo/configs/fullTest.conf]: Does it work? 🤔

Comment (VakarisZ) [reply envs/monkey_zoo/configs/fullTest.conf]: We don't have a vulnerable machine for it, but I added it for scanning purposes.

File: envs/monkey_zoo/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# MonkeyZoo
+These files are used to deploy Infection Monkey's test network on GCP.<br>
+For more information view MonkeyZooDocs.pdf
```
Comment (danielguardicore) [inline envs/monkey_zoo/README.md]: It's not a PDF anymore :)

File: envs/monkey_zoo/docs/fullDocs.md [hunk 4]
```diff
@@ -0,0 +1,1084 @@
...
+2.  Have a Google Cloud Platform account (upgraded if you want to test
+    whole network at once).
+
+To deploy:
+1.  Crete a service account for your project named
+    “your\_name-monkeyZoo-user” and download its **Service account
+    key**. Select JSON format.
+2.  Get these permissions in monkeyZoo project for your service account:
+
+    a.  **Compute Engine -\> Compute image user**
```
Comment (danielguardicore) [inline envs/monkey_zoo/docs/fullDocs.md]: Do we want to leave a note that using the TF script also requires

Compute Engine -\> Compute Network Admin
Compute Engine -\> Compute Instance Admin

File: envs/monkey_zoo/terraform/monkey_zoo.tf [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+      image = "${data.google_compute_image.hadoop-2.self_link}"
+    }
+    auto_delete = true
+  }
+  network_interface {
+    subnetwork="monkeyzoo-main"
+    network_ip="10.2.2.2"
+  }
+  // Add required ssh keys for hadoop service and restart it
+  metadata_startup_script = "[ ! -f /home/vakaris_zilius/.ssh/authorized_keys ] && sudo cat /home/vakaris_zilius/.ssh/id_rsa.pub >> /home/vakaris_zilius/.ssh/authorized_keys && sudo reboot"
```
Comment (danielguardicore) [inline envs/monkey_zoo/terraform/monkey_zoo.tf]: Uh what?

Comment (VakarisZ) [reply envs/monkey_zoo/terraform/monkey_zoo.tf]: Hadoop needs to be able to ssh into local host.

File: envs/monkey_zoo/docs/fullDocs.md (hunk 4 above)
Comment (VakarisZ) [reply envs/monkey_zoo/docs/fullDocs.md]: Hmm... Yeah, it's probably handy to leave what permissions you could give to the service account. If you're sure that those permissions are enough I'll add them.

File: envs/monkey_zoo/docs/fullDocs.md [hunk 6]
```diff
@@ -0,0 +1,1096 @@
...
+To deploy:
+1.  Configure service account for your project:
+
+    a. Create a service account and name it “your\_name-monkeyZoo-user” 
+    
+    b. Give these permissions to your service account:
+    
+    **Compute Engine -> Compute Network Admin**
+    and
+    **Compute Engine -> Compute Instance Admin**
```
Comment (danielguardicore) [inline envs/monkey_zoo/docs/fullDocs.md]: My mistake, we also require
Compute Engine -> Compute Network Admin
Compute Engine -> Service Account User

File: envs/monkey_zoo/terraform/monkey_zoo.tf (hunk 5 above)
Comment (danielguardicore) [reply envs/monkey_zoo/terraform/monkey_zoo.tf]: Why does it have your user?

Comment (VakarisZ) [reply envs/monkey_zoo/terraform/monkey_zoo.tf]: Hadoop is runing as my user and it need to ssh into localhost. When new VM instance is created (on monkeyZoo user for e.g.), all authorized_keys files are cleaned. We use this script to add those keys again.

**Pull request merged**